.github/workflows/check-performance-regression.yml

```
@@ -30,14 +30,16 @@ jobs:
           java-version: 11
           distribution: 'temurin'
 
-      - name: Checkout latest code
+      - name: Checkout Pull Request Code
         uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
```

@Lmh-java why do we need anything more than this:
```suggestion
      - name: checkout repo
        uses: actions/checkout@v4
```
?

---

Oh, I see, didn't realize that the action checkout the pull request by default.
as shown in https://github.com/actions/checkout.

---

Done

---

@Lmh-java I think we can assume that most modern CI will run on the reference it was triggered from unless otherwise specified.